lib/metasploit/framework/login_scanner/smb.rb

```diff
@@ -96,9 +96,9 @@ def attempt_login(credential)
           proof = nil
 
           begin
-            realm = (credential.realm || '').force_encoding('UTF-8')
-            username = (credential.public || '').force_encoding('UTF-8')
-            password = (credential.private || '').force_encoding('UTF-8')
+            realm = (credential.realm.dup || '').force_encoding('UTF-8')
+            username = (credential.public.dup || '').force_encoding('UTF-8')
+            password = (credential.private.dup || '').force_encoding('UTF-8')
```

Comment:
```suggestion
            realm = (credential.realm || '').dup.force_encoding('UTF-8')
            username = (credential.public || '').dup.force_encoding('UTF-8')
            password = (credential.private || '').dup.force_encoding('UTF-8')
```

There's the possibility that `''` is a frozen string in the future

Reply:
Ah, ok. I'll make that change 👍